gdal/port/cpl_string.h

```diff
@@ -334,20 +334,48 @@ extern "C++"
 #endif
 
 //! Convenient string class based on std::string.
```

Comment:
Please move that "//! Convenient string class based on std::string" Doxygen line just before "class CPL_DLL CPLString : public std::string", so as to fix the "cpl_string.h:366: warning: Compound CPLString is not documented." cppcheck error